src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java

```diff
@@ -2565,7 +2565,7 @@ protected void doMarshal(
                 Object source, HierarchicalStreamWriter writer, MarshallingContext context) {
             GeoServerInfo geoServerInfo = (GeoServerInfo) source;
             SettingsInfo settings = geoServerInfo.getSettings();
-            if (settings.isUseHeadersProxyURL() == null) {
+            if (settings != null && settings.isUseHeadersProxyURL() == null) {
```

Comment:
While we can add a null check; the real challenge is that during the load process settings should of been created.

I suspect that this is during the migration of an "old" data directory where only global.xml is defined....